### DISCUSSION TITLE
Is there a way to get the actual parser used by ESLint?

### DISCUSSION THREAD

**Discussion by ericmorand-sonarsource** [eslint/eslint#18965] [General] [answered]
Hey community,

We would like to write a rule that uses the same parser as the one used by ESLint to parse the currently analyzed file, but we are not sure how we can get that parser.

We were hoping that it would be available through the rule [Context](https://eslint.org/docs/latest/extend/custom-rules#the-context-object), but it seems like it is not: the only thing available is the  `languageOption.parser` value, which is not the parser used by ESLint but the value passed by the user when configuring ESLint; said differently, it may be a string.

Since ESLint already parsed the analyzed file prior to the rule execution, the parser was resolved and instantiated at that point. How can we retrieve it?

**Comment by mdjermanovic** [comment]
`context.languageOptions.parser` should always be the parser object that was used by ESLint for parsing that file. Can you provide an example where it is a string value?

**Comment by ericmorand-sonarsource** [comment]
The documentation says that it can be a string:

> parser: (object|string): Either the parser used to parse the current file for flat config or its name for legacy config.

In this case, we don't know how to retrieve the parser itself.

**Reply by nzakas** [reply, marked as answer]
It can be a string in the config object, but `context.languageOptions.parser` is always an object.

**Reply by ericmorand-sonarsource** [reply]
That's good news. I think the documentation needs to be updated, would you accept a PR about that?

**Reply by nzakas** [reply]
Documentation improvements are always welcome. :+1:

**Reply by ericmorand-sonarsource** [reply]
Awesome, thanks.

**Comment by ericmorand-sonarsource** [comment]
@nzakas I have one more question; according to the typings, `context.languageOptions.parser` may be `undefined`. Our testing seems to prove that even when the default ESLint parser is used (i.e., when no parser was set through the configuration), `context.languageOptions.parser` is defined as the default parser.

Can you confirm that the typings are wrong here?
